### PR TITLE
detect-flowbits: adding details for flowbits v5

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Task OISF#6309

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6309) ticket: https://redmine.openinfosecfoundation.org/issues/6309
Previous PR: https://github.com/OISF/suricata/pull/9691

Describe changes:
- added the recommended changes from the last PR.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1512

### CODE REVIEW

File: src/detect-engine-analyzer.c [hunk 1]
```diff
@@ -861,6 +863,45 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
+                if (cd->or_list_size == 0) {
+                    jb_append_string(js, VarNameStoreSetupLookup(cd->idx, VAR_TYPE_FLOW_BIT));
+                } else if (cd->or_list_size > 0) {
+                    for (uint8_t i = 0; i < cd->or_list_size; i++) {
+                        const char *varname =
+                                VarNameStoreSetupLookup(cd->or_list[i], VAR_TYPE_FLOW_BIT);
+                        jb_append_string(js, varname);
+                    }
+                }
+                jb_close(js); // array
```
Comment (jufajardini) [inline src/detect-engine-analyzer.c:901]: Considering that for `noalert` there won't be an associated variable name, I don't think we want to always open this array, only when it's not `noalert`, I guess. Not sure what's the best approach in terms of implementation, for this, right now.

Comment (jufajardini) [reply src/detect-engine-analyzer.c:901]: (btw, my guess is that that's what's causing the last check in the SV test to fail. this double close when nothing was added to the json object is leading to suricata not having the json object with the flowbits info, in the `noalert` case.)

File: src/detect-engine-analyzer.c [hunk 2]
```diff
@@ -861,6 +863,45 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
+                jb_close(js); // object
```
Comment (jufajardini) [inline src/detect-engine-analyzer.c:902]: Considering the discussion here - https://github.com/OISF/suricata/pull/9691#issuecomment-1783673052 -, we are missing the `operator` to be added to the details.

**Review by jufajardini — CHANGES_REQUESTED** [review]
We're getting there, I see that most SV checks are passing, now, kudos!

I've left some comments for the work to continue :)

Also please note that the commit message is still not following our guidelines.
Something like `detect/analyzer: add details to flowbits keyword` for the commit title would work. The ticket indication is good!

File: src/detect-engine-analyzer.c [hunk 3]
```diff
@@ -861,6 +863,45 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
                 break;
             }
+            case DETECT_FLOWBITS: {
+                const DetectFlowbitsData *cd = (const DetectFlowbitsData *)smd->ctx;
+
+                jb_open_object(js, "flowbits");
+                switch (cd->cmd) {
+                    case DETECT_FLOWBITS_CMD_NOALERT:
+                        jb_set_string(js, "action", "noalert");
+                        // jb_set_string(js,"name", NULL);
```
[the remark this answers is not in the document]
Comment (hadiqaalamdar) [reply src/detect-engine-analyzer.c:873]: yep, don't know how I missed that. I'll remove it right away.

File: src/detect-engine-analyzer.c (hunk 2 above)
Comment (hadiqaalamdar) [reply src/detect-engine-analyzer.c:902]: For this, I was thinking about creating an operator array that stores "or" if it encounters '|' in the the list, but I'm not sure on how to store "and" yet.

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10006

**Pull request closed without merging**